### PR TITLE
Implement add_space and get_spaces

### PR DESCRIPTION
**Pull request by gnuoy**
This PR implements add_space and get_spaces. Since spaces are not
implemented with the LXD provider there is not corresponding
functional test. Closes issue #466.

### CODE REVIEW

File: juju/client/connection.py [hunk 1]
```diff
@@ -99,7 +99,7 @@
     'RetryStrategy': {'versions': [1]},
     'Singular': {'versions': [2]},
     'SSHClient': {'versions': [1, 2]},
-    'Spaces': {'versions': [2, 3]},
+    'Spaces': {'versions': [2, 3, 6]},
```
Comment (SimonRichardson) [inline juju/client/connection.py:102]: Can you add, `4, 5` as well to this.

**Review by SimonRichardson — CHANGES_REQUESTED** [review]
@manadart can you check this as well 👍

**Review by SimonRichardson — APPROVED** [review]
Let's good from my perspective.

**Review by manadart — APPROVED** [review]
Looks good.

**Comment by SimonRichardson** [comment]
!!build!!

**Comment by SimonRichardson** [comment]
@gnuoy can you fix the lint issues, then I can land it.

> juju/tag.py:54:1: E302 expected 2 blank lines, found 1

**Comment by SimonRichardson** [comment]
!!build!!

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
$$merge$$

Failed to build python 3.5, considering it's EOL (#468) we should remove.

**Pull request merged**